### ISSUE TITLE
Will Google allow you to use this solution?

### ISSUE THREAD

**Issue by Serega007RU** [GrinZero/chrome-inject-eval#1]
Hello, your solution is impressive and working for Manifest V3. But how will Google react to the Chrome Web Store to such a trick when publishing an extension? Will it allow you to publish the extension? If yes, is there already a published extension with this solution?

**Comment by GrinZero** [comment]
As for the review of this situation, I didn't try to go online in the Google Store, so I'm sorry I can't tell you the answer. But I can tell you about similar situations I have encountered. In China, there is also a review of custom interpreters in WeChat applets. According to the solutions provided by the online giants, there are solutions that can be referred to (in essence, this method is to use a third-party JavaScript interpreter. In theory, as long as the Internet is available, nothing can be completely prohibited)

**Comment by Serega007RU** [comment]
Well, I'll probably be a pioneer, I'll finish migrating to manifest v3 and try to publish an extension with your solution. I'll let you know if it worked out or not

**Comment by GrinZero** [comment]
> Well, I'll probably be a pioneer, I'll finish migrating to manifest v3 and try to publish an extension with your solution. I'll let you know if it worked out or not

What about? Did you succeed in cheating

**Comment by Serega007RU** [comment]
Patience please :)
I am engaged in my project only in my free time, it may take several months. If Google really forbids running extensions on the manifest version 2 in 2023, then this will speed up the process. I won't forget and I'll let you know if it worked out or not.
If anything, here is my project https://gitlab.com/Serega007/Auto-Vote-Rating/-/tree/mv3, I have just started the transition to manifest version 3, but I have encountered several serious problems that will not be easy to solve

**Comment by Serega007RU** [comment]
It looks like I've run into a problem
I can't pass the function to chrome.scripting.executeScript()
On the page in the tab I always get this error
![image](https://user-images.githubusercontent.com/64780091/210229442-3d4e9778-65a5-49bc-bedf-06fe5572bb11.png)
Also, the code that comes is also kind of strange
![image](https://user-images.githubusercontent.com/64780091/210229494-f9bdaabf-f285-46fd-97de-bb899e7aa75a.png)
What I was trying to do
```JS
const func = new evalCore.evalModule.Function('console.log("test")')
chrome.scripting.executeScript({target: {tabId: 340788688}, func: func})
```

```JS
const instance = {}
const evil = evalCore.getEvalInstance(instance)
evil('function test() {console.log("test")}')
chrome.scripting.executeScript({target: {tabId: 340788688}, func: instance.test})
```

**Comment by GrinZero** [comment]
Here is an explanation of the last figure: you can try to bind the scope to the global, such as "globalThis"

**Comment by Serega007RU** [comment]
@GrinZero Hello! Google approved the update and published the extension. Here is the link https://chrome.google.com/webstore/detail/auto-vote-rating/mdfmiljoheedihbcfiifopgmlcincadd
I had to switch from gitlab to github to host the remote code, I did it because the domain gitlab.io blocked in Russia.
Here is a new link to the source code https://github.com/Serega007RU/Auto-Vote-Rating/
Regarding the error I encountered, I bypassed it and refused to use the functions. I'm doing an inject coreEval.js to the tab and already send the source code there via chrome.tabs.SendMessage
Thank you for your solution with the remote code problem on manifest v3
